src/main/java/pl/javastart/task/GrossPriceCalculator.java

```diff
@@ -0,0 +1,27 @@
+package pl.javastart.task;
+
+class GrossPriceCalculator {
+
+    double grossPrice(Product product) {
+        double universalVat = 28;
+        if (product.getCategory() != null) {
+            switch (product.getCategory().getName()) {
+                case "Słodycze":
+                case "Chemia":
+                    return grossPriceCalculator(product);
+                default:
+                    return grossPriceCalculator2(product, universalVat);
+            }
```

Comment:
Jeżeli kategoria jest obiektem, a nie Stringiem, to tego switcha można się u Ciebie pozbyć